langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/transport/http/SseEventListener.java

```diff
@@ -0,0 +1,64 @@
+package dev.langchain4j.mcp.client.transport.http;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import okhttp3.Response;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSourceListener;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SseEventListener extends EventSourceListener {
+
+    private final Map<Long, CompletableFuture<JsonNode>> pendingOperations;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger log = LoggerFactory.getLogger(SseEventListener.class);
+    private final boolean logEvents;
+
+    public SseEventListener(Map<Long, CompletableFuture<JsonNode>> pendingOperations, boolean logEvents) {
+        this.pendingOperations = pendingOperations;
+        this.logEvents = logEvents;
+    }
+
+    @Override
+    public void onClosed(EventSource eventSource) {
+        log.debug("SSE channel closed");
+    }
+
+    @Override
+    public void onEvent(EventSource eventSource, String id, String type, String data) {
+        if (type.equals("message")) {
+            if (logEvents) {
+                log.info("< {}", data);
+            }
+            try {
+                JsonNode message = OBJECT_MAPPER.readValue(data, JsonNode.class);
+                long messageId = message.get("id").asLong();
+                CompletableFuture<JsonNode> op = pendingOperations.remove(messageId);
+                if (op != null) {
+                    op.complete(message);
+                } else {
+                    log.warn("Received response for unknown message id: {}", messageId);
+                }
+            } catch (JsonProcessingException e) {
+                log.warn("Failed to parse response data", e);
+            }
+        }
+    }
+
+    @Override
+    public void onFailure(@NotNull final EventSource eventSource, final Throwable t, final Response response) {
```

Comment:
Hm my IDE generated this (the superclass is in Kotlin), I think we don't need it at all, do we..

Reply:
Not really, this will be called by okhttp, so why should we annotate?

But if we use those kind of annotations, let's use JSpecify which we just started using recently